crates/pallet-balanced-currency-swap-bridges-initializer/src/upgrade_init.rs

```diff
@@ -4,25 +4,31 @@ use frame_support::pallet_prelude::*;
 #[cfg(feature = "try-runtime")]
 use sp_std::vec::Vec;
 
-use crate::{Config, InitializerVersion, Pallet, CURRENT_BRIDGES_INITIALIZER_VERSION};
+use crate::{
+    Config, LastForceRebalanceAskCounter, LastInitializerVersion, Pallet,
+    CURRENT_BRIDGES_INITIALIZER_VERSION,
+};
 
-/// Initialize the bridges pot accounts.
+/// Initialize the bridges pot accounts if required.
 pub fn on_runtime_upgrade<T: Config>() -> Weight {
-    let initializer_version = <InitializerVersion<T>>::get();
-    let mut weight = T::DbWeight::get().reads(1);
-
-    if initializer_version != CURRENT_BRIDGES_INITIALIZER_VERSION {
-        let is_balanced = Pallet::<T>::is_balanced().unwrap_or_default();
-        weight += T::DbWeight::get().reads(8);
-
-        if !is_balanced {
-            match Pallet::<T>::initialize() {
-                Ok(w) => weight += w,
-                Err(err) => sp_tracing::error!("error during bridges initialization: {err:?}"),
-            }
+    let last_initializer_version = <LastInitializerVersion<T>>::get();
+    let last_force_rebalance_ask_counter = <LastForceRebalanceAskCounter<T>>::get();
+    let current_force_rebalance_ask_counter = T::ForceRebalanceAskCounter::get();
+
+    let mut weight = T::DbWeight::get().reads(3);
```

Comment:
```suggestion
    let mut weight = T::DbWeight::get().reads(2);
```

`T::ForceRebalanceAskCounter::get();` is not a db read